daemon/CryptoImpl/crypto.cpp

```diff
@@ -781,6 +825,73 @@ void Daemon::ApiImpl::CryptoRequestQueue::handlePendingRequest(
             }
             break;
         }
+        case ImportKeyRequest: {
+            qCDebug(lcSailfishCryptoDaemon) << "Handling GenerateKeyRequest from client:" << request->remotePid << ", request number:" << request->requestId;
+            Key importedKey;
+            Key key = request->inParams.size()
+                    ? request->inParams.takeFirst().value<Key>()
+                    : Key();
+            InteractionParameters uiParams = request->inParams.size()
+                    ? request->inParams.takeFirst().value<InteractionParameters>()
+                    : InteractionParameters();
+            QString cryptosystemProviderName = request->inParams.size()
+                    ? request->inParams.takeFirst().value<QString>()
```

Comment:
If we want to make such changes, let's do it in a single atomic PR that affects the entire codebase, rather than doing piecemeal modifications which cause inconsistency with the rest of the code.

I'm all for someone to refactor the code, but please do it atomically and globally.